util/syn_yosys.sh

```diff
@@ -18,31 +18,46 @@
 # use fusesoc to generate files and file list
 #-------------------------------------------------------------------------
 \rm -Rf build
-fusesoc --cores-root .. run --target=sim --setup --build formal > /dev/null 2>&1
+fusesoc --cores-root .. run --target=syn --setup lowrisc:systems:top_earlgrey
 
 # copy all files into directory "syn_out"
 \rm -Rf syn_out
 mkdir syn_out
-cp build/formal_0/src/*/*/*.sv build/formal_0/src/*/*/*/*.sv syn_out
+cp \
+  build/lowrisc_systems_top_earlgrey_0.1/src/*/*.sv \
+  build/lowrisc_systems_top_earlgrey_0.1/src/*/*/*.sv \
+  build/lowrisc_systems_top_earlgrey_0.1/src/*/*/*/*.sv \
+  syn_out
+cd syn_out
+
+# not synthesizable
+\rm prim_pad_wrapper.sv
+\rm prim_generic_pad_wrapper.sv
+
+# not used for synthesis
+\rm pins_if.sv
+\rm tlul_assert.sv
+\rm tlul_assert_multiple.sv
+
+# match filename to module name
+mv ibex_register_file{_ff,}.sv
+
+# prim_util_memload.sv is only meant to be included within a module
+mv prim_util_memload.sv{,h}
+sed -i.delete.bak -e "s/prim_util_memload\.sv/prim_util_memload.svh/" *.sv
+rm *.delete.bak  # some sed implementations require backup extensions
 
 #-------------------------------------------------------------------------
 # convert all RTL files to Verilog
 #-------------------------------------------------------------------------
-cd syn_out
-
-for file in *.sv; do
-  module=`basename -s .sv $file`
-  echo $file
-  sv2v --define=VERILATOR --define=SYNTHESIS *_pkg.sv prim_assert.sv $file > ${module}.v
-
-  # TODO: eventually remove below hack. It removes "unsigned" from params
-  # because Yosys doesn't support unsigned parameters
-  sed -i 's/parameter unsigned/parameter/g' ${module}.v
-  sed -i 's/localparam unsigned/localparam/g' ${module}.v
```

Comment:
Has this support been added to Yosys? (I couldn't find interesting occurrences of 'unsigned' in their testsuite, but that might just mean they don't have tests) or do we just not have any unsigned parameters in our codebase any more? If the former, it might be helpful to note the minimum version in the commit message?

Reply:
In IEEE 1364-2005, parameters are unsigned by default. While `parameter signed` is allowed, the grammar does not allow for the (redundant) `unsigned` keyword to be applied to parameters. I added a conversion in sv2v to remove these occurrences some time ago.

Reply:
Ah, right! Thanks for the explanation.